frontend/packages/dev-console/src/components/add/layout/Masonry.tsx

```diff
@@ -0,0 +1,72 @@
+import * as React from 'react';
+import * as cx from 'classnames';
+import Measure, { BoundingRect } from 'react-measure';
+import './MasonryLayout.scss';
+
+type MasonryProps = {
+  columnsCount: number;
+  children: React.ReactElement[];
+};
+
+type Size = BoundingRect;
+
+export const Masonry: React.FC<MasonryProps> = ({ columnsCount, children }) => {
+  const [heights, setHeights] = React.useState<Record<string, Size>>({});
+  const columns = columnsCount || 2;
+  const setHeight = (key: string, size: Size) => {
+    setHeights((old) => ({ ...old, [key]: size }));
+  };
```

Comment:
Also if we developer this together, I now thing we could simplify this and just save the height here.

```suggestion
  const [heights, setHeights] = React.useState<Record<string, number>>({});
  const columns = columnsCount || 1;
  const setHeight = (key: string, height: number) => {
    setHeights((old) => ({ ...old, [key]: height }));
  };
```